tests/Unit/IO/Observers/Test_RegisterSingleton.cpp

```diff
@@ -0,0 +1,116 @@
+  struct Result {
+    observers::ObservationKey observation_key{};
+    size_t caller_node_id{};
+  };
+  static Result result;
+
+  template <typename ParallelComponent, typename DbTagList,
+            typename Metavariables, typename ArrayIndex>
+  static void apply(db::DataBox<DbTagList>& /*box*/,
+                    Parallel::GlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const observers::ObservationKey& observation_key,
+                    const size_t caller_node_id) noexcept {
+    result.observation_key = observation_key;
+    result.caller_node_id = caller_node_id;
+  }
+};
+
+MockRegisterReductionContributorWithObserverWriter::Result
+    MockRegisterReductionContributorWithObserverWriter::result{};
+
+template <typename Metavariables>
+struct MockObserverWriterComponent {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using const_global_cache_tags = tmpl::list<>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<typename Metavariables::Phase,
+                             Metavariables::Phase::Testing, tmpl::list<>>>;
+
+  using component_being_mocked = observers::ObserverWriter<Metavariables>;
+  using replace_these_simple_actions =
+      tmpl::list<observers::Actions::RegisterReductionNodeWithWritingNode>;
+  using with_these_simple_actions =
+      tmpl::list<MockRegisterReductionContributorWithObserverWriter>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<Component<Metavariables>,
+                                    MockObserverWriterComponent<Metavariables>>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+SPECTRE_TEST_CASE("Unit.IO.Observers.RegisterSingleton", "[Unit][Observers]") {
+  using my_component = Component<Metavariables>;
+  using obs_component = MockObserverWriterComponent<Metavariables>;
+  ActionTesting::MockRuntimeSystem<Metavariables> runner{{}};
+  ActionTesting::emplace_component<my_component>(&runner, 0);
+  ActionTesting::emplace_component<my_component>(&runner, 1);
+  ActionTesting::emplace_component<obs_component>(&runner, 0);
+  ActionTesting::set_phase(make_not_null(&runner),
+                           Metavariables::Phase::Testing);
+
+  ActionTesting::next_action<my_component>(make_not_null(&runner), 0);
+  REQUIRE(not ActionTesting::is_simple_action_queue_empty<obs_component>(runner,
+                                                                         0));
+  ActionTesting::invoke_queued_simple_action<obs_component>(
+      make_not_null(&runner), 0);
+  REQUIRE(
+      ActionTesting::is_simple_action_queue_empty<obs_component>(runner, 0));
+
+  CHECK(MockRegisterReductionContributorWithObserverWriter::result
+            .observation_key == observers::ObservationKey{"Singleton"});
+  // Need an `or` because we don't know what order actions are run in
```

Comment:
Is this comment from a prior code version? I can't figure out what the "or" is that's being talked about.